docs/painless/painless-contexts/painless-bucket-script-agg-context.asciidoc

```diff
@@ -0,0 +1,87 @@
+[[painless-bucket-script-agg-context]]
+=== Bucket Script aggregation context
+
+Use a Painless script in an
+{ref}/search-aggregations-pipeline-bucket-script-aggregation.html[`bucket_script` pipeline aggregation]
+to calculate a value as a result in a bucket.
+
+==== Variables
+
+`params` (`Map`, read-only)::
+        User-defined parameters passed in as part of the query. The parameters
+        include values defined as part of the `buckets_path`.
+
+==== Return
+
+numeric::
+        The calculated value as the result.
+
+==== API
+
+The standard <<painless-api-reference, Painless API>> is available.
+
+==== Example
+
+To run this example, first follow the steps in <<painless-context-examples, context examples>>.
+
+The painless context in a `bucket_script` aggregation provides a `params` map.  This map contains both
+user-specified custom values, as well as the values from other aggregations specified in the `buckets_path`
+property.
+
+For our example, we are going to take the values from a `min` and `max` aggregation, subtract them to find
+the difference, then add in a "base cost" from a user-specified value.  The script looks like this:
+
```

Comment:
In general, I recommend avoiding the use of "we" or "our". In this case, I'd just say:

This example takes the values from a `min` and `max` aggregation, calculates the difference, and adds the user-specified `base_cost` to the result.